app/bundles/LeadBundle/Tests/EventListener/Issue9488Test.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+namespace Mautic\LeadBundle\Tests\EventListener;
+
+use Mautic\CampaignBundle\Entity\Campaign;
+use Mautic\CampaignBundle\Entity\Event;
+use Mautic\CampaignBundle\Entity\Lead as CampaignLead;
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Entity\LeadRepository;
+use PHPUnit\Framework\Assert;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Component\Console\Tester\ApplicationTester;
+use Symfony\Component\HttpFoundation\Response;
+
+class Issue9488Test extends MauticMysqlTestCase
+{
+    private LeadRepository $contactRepository;
+
+    /**
+     * @var array<int, array<string, string|int>>
+     */
+    private array $contacts = [
+        [
+            'email'     => 'contact1@email.com',
+            'firstname' => 'Isaac',
+            'lastname'  => 'Asimov',
+        ],
+        [
+            'email'            => 'contact2@email.com',
+            'firstname'        => 'Robert A.',
+            'lastname'         => 'Heinlein',
+            'points'           => 0,
+            'preferred_locale' => 'af',
+        ],
+        [
+            'email'     => 'contact3@email.com',
+            'firstname' => 'Arthur C.',
+            'lastname'  => 'Clarke',
+            'points'    => 1,
+        ],
+    ];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->contactRepository = $this->em->getRepository(Lead::class);
```

Comment:
Try if PHPSTAN can work with this:
```suggestion
        /** @var LeadRepository $contactRepository
        $contactRepository = $this->em->getRepository(Lead::class);

        $this->contactRepository = $contactRepository;
```

Reply:
Sure that works. I thought you are adding more items to the ignore file so you will have all "wrong pieces" in one place for the overview and the fixup later, and not in the phpdoc.

Reply:
That would make sense, but the history is a bit different. We used to use PHPSTAN on level 1 (the lowest). One day we were hunting a bug and found out that PHPSTAN on a higher level would have discovered it. So we wanted to bump to this higher level so the bug and all the similar bugs would have been caught in the CI. But it would take months to fix all the [existing PHPSTAN issues.](https://github.com/mautic/mautic/blob/5.x/phpstan-baseline.neon) As no one would fund such effort, PHPSTAN has this handy [Baseline](https://phpstan.org/user-guide/baseline) feature that allowed us to push all the tech debt into 1 file and use higher level on the newly added/modified code.